src/main/java/com/rapid7/jul/LogentriesHandler.java

```diff
@@ -1,112 +1,76 @@
 
-/**
- * <code>LogentriesHandler</code>: A handler for writing formatted records to a
- * logentries.com. This handler uses the Token-based input.
- *
- * @author Björn Raupach (raupach@me.com)
- */
-public final class LogentriesHandler extends Handler {
+public class LogentriesHandler extends Handler {
 
-    private final byte[] newline = {0x0D, 0x0A};
-    private final byte space = 0x020;
-    private String host;
-    private int port;
-    private byte[] token;
-    private String region;
-    private boolean open;
-    private SocketChannel channel;
-    private ByteBuffer buffer;
+    /**
+     * Asynchronous Background logger
+     */
+    AsyncLogger iopsAsync;
 
     public LogentriesHandler() {
         this(null);
     }
 
     public LogentriesHandler(String prefix) {
+        iopsAsync = new AsyncLogger();
         configure(prefix);
-        connect();
-        buffer = ByteBuffer.allocate(4096);
     }
 
-    public String getHost() {
-        return host;
+    void configure(String prefix) {
+        String cname = getClass().getName();
+        String propsPrefix = prefix == null ? cname : prefix + "." + cname;
+        setLevel(getLevelProperty(propsPrefix + ".level", Level.INFO));
+        setFormatter(getFormatterProperty(propsPrefix + ".formatter", new SimpleFormatter()));
+        setRegion(getStringProperty(propsPrefix + ".region", ""));
+        setHost(getStringProperty(propsPrefix + ".host", null));
+        setPort(getIntProperty(propsPrefix + ".port", 0));
+        setToken(getStringProperty(propsPrefix + ".token", ""));
+        setUseDataHub(getBooleanProperty(propsPrefix + ".useDataHub", false));
+        setHttpPut(getBooleanProperty(propsPrefix + ".httpPut", false));
+        setSsl(getBooleanProperty(propsPrefix + ".ssl", true));
     }
 
-    public void setHost(String host) {
-        this.host = host;
+    private void setUseDataHub(boolean useDataHub) {
+        iopsAsync.setUseDataHub(useDataHub);
     }
 
-    public int getPort() {
-        return port;
+    private void setRegion(String region) {
+        iopsAsync.setRegion(region);
     }
 
-    public void setPort(int port) {
-        this.port = port;
-    }
 
-    public byte[] getToken() {
-        return token;
+    private void setHost(String host) {
+        iopsAsync.setDataHubAddr(host);
     }
 
-    public void setToken(byte[] token) {
-        this.token = token;
+    private void setPort(int port) {
+        iopsAsync.setDataHubPort(port);
     }
 
-    public String getRegion() {
-        return region;
+    private void setToken(String token) {
+        iopsAsync.setToken(token);
     }
 
-    public void setRegion(String region) {
-        this.region = region;
+    public void setSsl(boolean ssl) {
+        this.iopsAsync.setSsl(ssl);
     }
 
+    public void setHttpPut(boolean httpPut) {
+        this.iopsAsync.setHttpPut(httpPut);
+    }
 
     @Override
-    public synchronized void publish(LogRecord record) {
```

Comment:
Several methods have been removed from this class. Are they no longer needed?

Reply:
yes, because there is not need anymore to manage the buffer and the stream in this class. That logic is delegated to LogEntriesHandler and AsyncLogger